test/with_api_v2/acceptance.go

```diff
@@ -252,16 +256,45 @@ type Alertmanager struct {
 	errc chan<- error
 }
 
+// AlertmanagerCluster represents a group of Alertmanager instances
+// acting as a cluster.
+type AlertmanagerCluster struct {
+	ams []*Alertmanager
+}
+
+// Start the Alertmanager cluster and wait until it is ready to receive.
+func (amc *AlertmanagerCluster) Start() {
+	var peerFlags []string
+	for _, am := range amc.ams {
+		peerFlags = append(peerFlags, "--cluster.peer="+am.clusterAddr)
+	}
+
+	for _, am := range amc.ams {
+		am.Start(peerFlags)
+	}
+
+	for _, am := range amc.ams {
+		err := am.WaitForCluster(len(amc.ams))
+		if err != nil {
+			// TODO: Properly handle this error
+			fmt.Println(err)
+		}
+	}
+}
+
 // Start the alertmanager and wait until it is ready to receive.
-func (am *Alertmanager) Start() {
-	cmd := exec.Command("../../../alertmanager",
+func (am *Alertmanager) Start(additionalArg []string) {
+	args := []string{
 		"--config.file", am.confFile.Name(),
 		"--log.level", "debug",
 		"--web.listen-address", am.apiAddr,
 		"--storage.path", am.dir,
 		"--cluster.listen-address", am.clusterAddr,
 		"--cluster.settle-timeout", "0s",
-	)
+	}
+	args = append(args, additionalArg...)
+
+	cmd := exec.Command("../../../alertmanager", args...)
```

Comment:
It's been mentioned before, but it would be good to reorganize the main code so it can be executed in the test, instead of relying on the environment executing this test to have a binary, and that it's been built off the current code.

Reply:
(just a note for the future)

Reply:
Yes, that would also speed up the tests by a lot. Will look into that soonish.